### PR TITLE
chore: config junit5 and samples for project unit testing

### PR DESCRIPTION
**Pull request by yamilmedina**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [X] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [X] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [X] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

Currently, as we discussed on the android collective, we are not covering most of the components on the project.
So this adds:  
- The basic configuration needed to run tests using Junit5. 
- A sample tests for a simple extension function on `DateUtils` -
- A sample test coverage `WelcomeViewModel`, this ViewModel depends on a Junit extension (former @Rule on Junit4) that will set up the Dispatchers to `TestCoroutineDispatcher` for the coroutine work to be executed for tests before and after every run

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [X] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [X] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: .github/workflows/gradle-run-ui-tests.yml [hunk 1]
```diff
@@ -50,15 +50,13 @@ jobs:
...
       - name: Android Instrumentation Tests
         uses: reactivecircus/android-emulator-runner@v2
         with:
           api-level: {{ matrix.api-level }}
           script: ./gradlew connectedCheck
 
+        # TODO: verify route for android tests
       - name: Publish Unit Test Results
         uses: EnricoMi/publish-unit-test-result-action/composite@v1.25
         if: always()
```
Comment (gongracr) [inline .github/workflows/gradle-run-ui-tests.yml:62]: If above says UI tests, why are we running also the publishing of the unit tests in this action?

File: app/src/test/kotlin/com/wire/android/util/DateTimeUtilKtTest.kt [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+class DateTimeUtilKtTest {
+
+    @Test
+    fun `given an invalid date, when performing a transformation, then return null`() {
+        val result = "NOT_VALID".formatMediumDateTime()
+        assertEquals(null, result)
+    }
+
+    @Test
+    fun `given an valid date, when performing a transformation, then return with medium format`() {
```
Comment (gongracr) [inline app/src/test/kotlin/com/wire/android/util/DateTimeUtilKtTest.kt:15]:
```suggestion
    fun `given a valid date, when performing a transformation, then return with medium format`() {
```

**Review by gongracr — COMMENTED** [review]
Our first view model tests looking real good 🍍 ! Does this also mean that we will be running UI and unit tests in parallel?

File: app/src/test/kotlin/com/wire/android/config/CoroutineTestExtension.kt [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+ *
+ * Annotating the class with
+ * @ExtendWith(CoroutineTestExtension::class)
+ */
+@ExperimentalCoroutinesApi
+class CoroutineTestExtension(
+    private val dispatcher: TestCoroutineDispatcher = TestCoroutineDispatcher()
+) : BeforeEachCallback, AfterEachCallback, TestCoroutineScope by TestCoroutineScope(dispatcher) {
+
+    override fun beforeEach(context: ExtensionContext?) {
```
Comment (gongracr) [inline app/src/test/kotlin/com/wire/android/config/CoroutineTestExtension.kt:29]: How is this context being used?

File: .github/workflows/gradle-run-ui-tests.yml (hunk 1 above)
Comment (yamilmedina) [reply .github/workflows/gradle-run-ui-tests.yml]: In fact, it's a leftover from the original file, now we are going to run `unit test` and `ui test` on different workflows, achieving parallelism =)

File: app/src/test/kotlin/com/wire/android/config/CoroutineTestExtension.kt (hunk 3 above)
Comment (yamilmedina) [reply app/src/test/kotlin/com/wire/android/config/CoroutineTestExtension.kt:29]: In fact this it's not used (we don't need it) we are just overriding the `beforeEach`  and `afterEach` from the contracts `BeforeEachCallback` and `AfterEachCallback` interfaces, to set and reset the dispatcher to be used on testing context

**Review by gongracr — APPROVED** [review]

**Review by ohassine — APPROVED** [review]
👌

**Pull request merged**